### PR TITLE
Adelle/contextual sidebar

### PR DESCRIPTION
**Pull request by Adelle-Pitsas**
Makes sidebar contextual based on what node you click on:

- click on operation, shows info on operation (already done)
- click on service, shows info on service
- click on component (Kafka) shows info on kafka (and some metrics to come later)
- click on background and it shows empty state

Currently since you can drag from any part of the node, I kept the grab cursor, but since you can also now click on any node, its a bit awkward and unintuitive to a user that you could click on it and get more info (aka grabbing and clicking happens in the same space). Happy to change it but unsure what to change it to.

https://github.com/streamdal/snitch-console/assets/108096652/e9e2c0f1-8a48-4ca2-ad0b-6154372cc9d3

### CODE REVIEW

File: islands/opModal.tsx [hunk 1]
```diff
@@ -36,10 +38,41 @@ export default function OpModal(
     grpcToken: string;
   },
 ) {
+  const displayType = opModal.value?.displayType;
+  const setDisplayName = () => {
```
Comment (jacobheric) [inline islands/opModal.tsx:42]: I think this should be named get not set as it's not setting anything. Also, a switch case might be a bit more readable than if/else

File: islands/opModal.tsx [hunk 2]
```diff
@@ -36,10 +38,41 @@ export default function OpModal(
...
+    if (displayType === "operation") {
+      return opModal.value?.audience?.operationName;
+    } else if (displayType === "service") {
+      return opModal.value?.audience?.serviceName;
+    } else if (displayType === "component") {
+      return opModal.value?.audience.componentName;
+    }
+  };
+
+  const iconDisplay = () => {
```
Comment (jacobheric) [inline islands/opModal.tsx:52]: If this display function logic gets re-used across the nodes and drawer you could make them react components.

File: islands/opModal.tsx [hunk 3]
```diff
@@ -136,270 +141,281 @@ export default function OpModal(
...
-                            <a href={"/pipelines"}>
-                              <button class="text-streamdalPurple border border-purple-600 bg-purple-50 font-semibold rounded-lg w-full flex justify-center text-sm px-2 py-1 text-center inline-flex items-center">
-                                <IconPlus class="w-4 h-4 mr-1" />
-                                Create a new pipeline
-                              </button>
-                            </a>
-                          )
-                          : attachedPipeline
-                          ? (
+                      {displayType === "operation"
```
Comment (jacobheric) [inline islands/opModal.tsx:159]: Might make sense to break out these drawer parts into different sub-components for operation versus other types remove some of these ternaries.

**Review by jacobheric — APPROVED** [review]
Some stylistic feedback but otherwise lgtm!

File: islands/opModal.tsx (hunk 3 above)
Comment (Adelle-Pitsas) [reply islands/opModal.tsx:159]: Sounds good. I'm going to merge this in and handle breaking out the opModal component in a different branch.

**Pull request merged**